+sw_tests/+unit_tests/unittest_spinw_genmagstr.m

```diff
@@ -0,0 +1,540 @@
+            testCase.verify_obj(expected_mag_str, swobj.mag_str);
+        end
+        function test_tile_input_S_extend_cell(testCase)
+            % Test that tile and input S less than nExt will correctly tile
+            % input S
+            swobj = copy(testCase.swobj);
+            swobj.addatom('r', [0.5 0.5 0], 'S', 1);
+            nExt = [int32(3) int32(1) int32(1)];
+            swobj.genmagstr('mode', 'tile', 'nExt', nExt, ...
+                            'S', [1 0; 0 1; 0 0]);
+            expected_mag_str = testCase.default_mag_str;
+            expected_mag_str.nExt = nExt;
+            expected_mag_str.F = [1 0 1 0 1 0; 0 1 0 1 0 1; 0 0 0 0 0 0];
+            testCase.verify_obj(expected_mag_str, swobj.mag_str);
+        end
+        function test_tile_multik(testCase)
+            % Test that S is summed over third dimension with tile, and k
+            % is not needed (is this the behaviour we want?)
+            swobj = copy(testCase.swobj);
+            swobj.addatom('r', [0.5 0.5 0], 'S', 1);
+            S = cat(3, [1 0; 0 1; 0 0], [0 1; 0 0; 1 0]);
+            swobj.genmagstr('mode', 'tile', ...
+                            'S', S);
+            expected_mag_str = testCase.default_mag_str;
+            expected_mag_str.F = sqrt(2)/2*[1 1; 0 1; 1 0];
+            testCase.verify_obj(expected_mag_str, swobj.mag_str);
+        end
+        function test_tile_multik_provided_k_set_to_zero(testCase)
+            % Test that S is summed over third dimension with tile, and if
+            % k is provided, it is set to zero anyway
+            swobj = copy(testCase.swobj);
+            swobj.addatom('r', [0.5 0.5 0], 'S', 1);
+            S = cat(3, [1 0; 0 1; 0 0], [0 1; 0 0; 1 0]);
+            k = 0.5*ones(size(S, 3), 3);
+            swobj.genmagstr('mode', 'tile', ...
+                              'S', S, 'k', k);
+            expected_mag_str = testCase.default_mag_str;
+            expected_mag_str.F = sqrt(2)/2*[1 1; 0 1; 1 0];
+            testCase.verify_obj(expected_mag_str, swobj.mag_str);
+        end
+        function test_extend_mode_input_S_extend_cell_and_warns(testCase)
+            % Test undocumented 'extend' mode does same as tile
+            % Test that tile and input S less than nExt will correctly tile
+            % input S
+            swobj = copy(testCase.swobj);
+            swobj.addatom('r', [0.5 0.5 0], 'S', 1);
+            nExt = [int32(3) int32(1) int32(1)];
+            testCase.verifyWarning(...
+                @() swobj.genmagstr('mode', 'extend', 'nExt', nExt, ...
+                                    'S', [1 0; 0 1; 0 0]), ...
+                'spinw:genmagstr:DeprecationWarning');
+            expected_mag_str = testCase.default_mag_str;
+            expected_mag_str.nExt = nExt;
+            expected_mag_str.F = [1 0 1 0 1 0; 0 1 0 1 0 1; 0 0 0 0 0 0];
+            testCase.verify_obj(expected_mag_str, swobj.mag_str);
+        end
+        function test_rotate_phi(testCase)
```

Comment:
`test_rotate_phi` and `test_rotate_phid` cold probably be combined into one parameterized test - also is it necessary to test what happens when bothy `phid` and `phi` are provided - or is that overkill?

Reply:
Done, and added `phi` + `phid` test because it was easy enough